datafusion/core/src/physical_plan/file_format/mod.rs

```diff
@@ -234,6 +234,39 @@ impl FileScanConfig {
     }
 }
 
+impl Display for FileScanConfig {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        let (schema, _, ordering) = self.project();
+
+        let output_ordering_str = ordering
+            .map(|v| make_output_ordering_string(&v))
+            .unwrap_or(String::from("[]"));
+
+        write!(
+            f,
+            "file_groups={}, projection={}, limit={:?}, output_ordering={}",
+            FileGroupsDisplay(&self.file_groups),
+            ProjectSchemaDisplay(&schema),
+            self.limit,
+            output_ordering_str
+        )
+    }
+}
+
+fn make_output_ordering_string(ordering: &[PhysicalSortExpr]) -> String {
```

Comment:
I think you could avoid the extra copy / materialization by doing something like

```rust
struct DisplayableOrdering( &[PhysicalSortExpr]);

impl Display for DisplayableOrdering {
    fn fmt(&self, f: &mut Formatter) -> FmtResult {
       // use self.0 here and call write! to f
    } 
}
```


Alternately you could probably use join here instead well: https://doc.rust-lang.org/std/primitive.slice.html#method.join
https://doc.rust-lang.org/std/primitive.slice.html#method.join

Reply:
Thanks I made the change with a wrapper.

However, I kept the imperative style instead of using join as I think join will do the actual copy, though I don't think performance matters that much for such path though.